### PR TITLE
Automation id for choice group options

### PR DESCRIPTION
**Pull request by antonlabunets**
#### Pull request checklist

- [V] Include a change request file using `$ npm run change`

#### Description of changes

Changed ChoiceGroupOption to inherit from input attributes such that additional attributes like data-automation-id can be added.

#### Focus areas to test
n/a
(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.Props.ts [hunk 1]
```diff
@@ -107,4 +107,9 @@ export interface IChoiceGroupOption {
    */
   labelId?: string;
 
+  /**
+   * Optional identifier for automating the clickable control in the choice field.
+   */
+  automationId?: string;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.Props.ts]: data-automation-id?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.Props.ts]: Or better, can we extend the interface from HTMLAttribute<something> and mix in the extra things? That way we can support passing in data-automation-id.

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.Props.ts [hunk 2]
```diff
@@ -107,4 +107,11 @@ export interface IChoiceGroupOption {
    */
   labelId?: string;
 
+  /**
+  * Optional input props that will be mixed into the input element, *before* other props are applied. This allows
+  * you to extend the input element with additional attributes, such as data-automation-id needed for automation.
+  * Note that if you provide, for example, "disabled" as well as "inputProps.disabled", the former will take
+  * precedence over the later.
+  */
+  inputProps?: React.HTMLAttributes<HTMLElement | HTMLInputElement>;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.Props.ts]: I'm a bit confused; why do we need inputProps? You're extending the input element interface, so I would suspect that we would just extract the input props from the option itself.

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx [hunk 3]
```diff
@@ -100,6 +102,7 @@ export class ChoiceGroup extends BaseComponent<IChoiceGroupProps, IChoiceGroupSt
                 }
               >
                 <input
+                  { ...getNativeProps(option.inputProps, inputProperties) }
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: Shouldn't this just be `getNativeProps(option, inputProperties)` ?

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.Props.ts (hunk 2 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.Props.ts]: This appears to mirror how we're doing it for checkbox. Seems we should determine the proper way to approach this, and get it into docs (and update checkbox as well)

https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.Props.ts#L63

**Comment by kristoferbrown** [comment]
@micahgodbolt, @dzearing, @antonlabunets, any chance we can schedule something to decide how to handle this and checkbox?

**Comment by micahgodbolt** [comment]
@kristoferbrown yeah, an input component should just extending React.HTMLAttributes<HTMLElement | HTMLInputElement> rather than placing them into a single prop. We should go back and update the checkbox as well. 

@antonlabunets are you able to go back make the change to the choicegroup?

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx (hunk 3 above)
Comment (antonlabunets) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: Adjusted.

**Comment by antonlabunets** [comment]
Adjusted the change as requested by @micahgodbolt

**Comment by dzearing** [comment]
@christiango For nullable types, what is the right fix here? Take a look at the build errors in travis; what's the right recommendation for nullables? In this case, @antonlabunets extended HTMLAttributes, which ends up causing some nullable errors downstream. If we just update the tests, that implies the this change could cause compiler errors for partners and that they have to do the same workaround.

**Pull request merged**